skaffold/src/main/kotlin/com/google/container/tools/skaffold/run/SkaffoldCommandLineState.kt

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.container.tools.skaffold.run
+
+import com.google.container.tools.skaffold.SkaffoldExecutorService
+import com.google.container.tools.skaffold.SkaffoldExecutorSettings
+import com.google.container.tools.skaffold.message
+import com.intellij.execution.ExecutionException
+import com.intellij.execution.configurations.CommandLineState
+import com.intellij.execution.process.KillableProcessHandler
+import com.intellij.execution.process.ProcessHandler
+import com.intellij.execution.runners.ExecutionEnvironment
+import com.intellij.openapi.vfs.LocalFileSystem
+import com.intellij.openapi.vfs.VfsUtilCore
+import java.io.File
+
+/**
+ * Generic [CommandLineState] implementation for running Skaffold on command line based on the given
+ * [AbstractSkaffoldRunConfiguration] and current IDE project. [startProcess] checks configuration
+ * and constructs command line to launch Skaffold process. Base class manages console
+ * window output and graceful process shutdown (also see [KillableProcessHandler])
+ *
+ * @param environment Execution environment provided by IDE
+ * @param executionMode Skaffold  mode execution, i.e. DEV
+ */
+class SkaffoldCommandLineState(
+    environment: ExecutionEnvironment,
+    val executionMode: SkaffoldExecutorSettings.ExecutionMode
+) : CommandLineState(environment) {
+    public override fun startProcess(): ProcessHandler {
+        val runConfiguration = environment.runnerAndConfigurationSettings?.configuration
+        val projectBaseDir = environment.project.baseDir
+        // ensure the configuration is valid for execution - settings are of supported type,
+        // project is valid and Skaffold file is present.
+        if (runConfiguration == null || runConfiguration !is AbstractSkaffoldRunConfiguration ||
+            projectBaseDir == null
+        ) {
+            throw ExecutionException(message("skaffold.unsupported.run.settings"))
```

Comment:
under what circumstances would this message be shown to the user? is it possible to be more specific with the message? As is, I don't know if I would know how to start resolving it if I saw this message.

Reply:
This is going to be shown when you click `Run` and something is very wrong, i.e. mismatch in configuration type or empty project directory etc. Very unlikely to happen, probably in case of some bad migration or issue. What about `Skaffold run configuration is invalid. Please edit to fix errors, or re-create new one.`?

Reply:
I wonder if its so rare/bad of a situation that it warns an IDE exception? If not, we could say something along the lines of `Your Skaffold run configuration is corrupted` followed by some wording to recreate a new one like you suggested.

Not to self: we need to add the plumbing for IDE exception reporting like with have in CT4IJ. I'll create an issue.

Reply:
Hard to say really, it looks like it could only happen in case of some deep state corruption. So you suggest we throw some sort of `RuntimeException` here to invoke IDE reporting mechanics, correct?

Reply:
I think I'm ok in this situation just notifying the user that their state is corrupted and to re-create the run configuration.

(btw if we had gone the route of triggering an IDE error, `log.error()` would have worked too)